src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1089,6 +1089,68 @@ internal void SelectAllCategories(object parameter)
             }
         }
 
+        /// <summary>
```

Comment:
@ramramps 
In any case, if you want to find something, then you need to search for something. If we want to set false for all expanded items, we must search for these expanded items. And in this case we will search among all items just  for expanded items. So, in any case we need to traverse all categories.
But! Another question: what is faster assignment or an if-statement?
http://stackoverflow.com/questions/4043455/which-costs-more-while-looping-assignment-or-an-if-statement

I decided to check :)
* Without if
search for point:
![image](https://cloud.githubusercontent.com/assets/8158404/12199937/7ea0a1aa-b624-11e5-8493-9c3c5b85be36.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12199944/890c07e2-b624-11e5-956e-f163b5492e9f.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12199953/94a79d64-b624-11e5-864a-821cf8d09a3e.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12199959/9fcd248e-b624-11e5-826a-d8e1f4dcbfef.png)

* With if
search for point:
![image](https://cloud.githubusercontent.com/assets/8158404/12199984/c133204c-b624-11e5-92e0-e75ad295353b.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12199991/cabec8a0-b624-11e5-9fda-aac2caa46166.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12199997/d37d12c6-b624-11e5-9e84-b3f56b5e9870.png)
![image](https://cloud.githubusercontent.com/assets/8158404/12200007/e113315e-b624-11e5-9b77-95c430f4564a.png)

I can't say, that I see big lose of time. So, I believe we can leave without if.

P.S. that was really interesting research :D